tests/integration/devfile/cmd_devfile_list_test.go

```diff
@@ -1,26 +1,21 @@
 package devfile
 
 import (
-	"fmt"
+	"path"
 	"path/filepath"
 
-	devfilepkg "github.com/devfile/api/v2/pkg/devfile"
-	"github.com/redhat-developer/odo/pkg/component"
 	"github.com/redhat-developer/odo/tests/helper"
-	"github.com/tidwall/gjson"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 )
 
 var _ = Describe("odo list with devfile", func() {
```

Comment:
I would probably go about structuring this test like [this](https://gist.github.com/valaparthvi/1161e562b04148bed555beb38c9911a9).

Reply:
Had to structure it similar to `odo push` and `odo dev` so it'd perform the correct tests with out-of-context-dir testing

Reply:
But in that case, you can simply add a `When("executing odo list outside the context directory,func(){})` block to the structure that I've suggested.